integration_test/ops_agent_test.go

```diff
@@ -1893,6 +1901,153 @@ func TestPrometheusMetrics(t *testing.T) {
 		}
 	})
 }
+
+// Test the Counter and Gauge metric types using a JSON Prometheus exporter
+// The JSON exporter will connect to a Python http server that serve static JSON files
+func TestPrometheusMetricsWithJSONExporter(t *testing.T) {
+	t.Parallel()
+	gce.RunForEachPlatform(t, func(t *testing.T, platform string) {
+		t.Parallel()
+		if gce.IsWindows(platform) {
+			t.SkipNow()
+		}
+		ctx, logger, vm := agents.CommonSetup(t, platform)
+
+		type fileToUpload struct {
+			local, remote string
+		}
+
+		filesToUpload := []fileToUpload{
+			{local: "testdata/data.json",
+				remote: "data.json"},
+			{local: "testdata/json_exporter_config.yaml",
+				remote: "json_exporter_config.yaml"},
+		}
+
+		workDir := path.Join(workDirForPlatform(vm.Platform), "json_exporter")
+
+		for _, file := range filesToUpload {
+			f, err := os.Open(file.local)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = gce.UploadContent(ctx, logger, vm, f, path.Join(workDir, file.remote))
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		packages := []string{"wget", "python3"}
+		err := agents.InstallPackages(ctx, logger.ToMainLog(), vm, packages)
+		if err != nil {
+			t.Fatalf("failed to install %v with err: %s", packages, err)
+		}
+
+		// Install either tmux or screen to run the JSON exporter and the http server in the background
+		tmux_err := agents.InstallPackages(ctx, logger.ToMainLog(), vm, []string{"tmux"})
+		if tmux_err != nil {
+			screen_err := agents.InstallPackages(ctx, logger.ToMainLog(), vm, []string{"screen"})
+			if screen_err != nil {
+				t.Fatalf("failed to install either tmux or screen with err: %s and %s", tmux_err, screen_err)
+			}
+		}
+
+		// Run the setup script to run the Python http server and the JSON exporter using tmux/screen
+		setupScript, err := os.ReadFile("testdata/setup_json_exporter.sh")
+
+		if err != nil {
+			t.Fatalf("failed to open setup script: %s", err)
+		}
+
+		env := map[string]string{"WORKDIR": workDir}
+		test_out, err := gce.RunScriptRemotely(ctx, logger, vm, string(setupScript), nil, env)
+
+		if err != nil {
+			t.Fatalf("failed to run json exporter in VM with err: %v, stderr: %s", err, test_out.Stderr)
+		}
+
+		config := `metrics:
+  receivers:
+    prom_app:
+      type: prometheus
+      config:
+        scrape_configs:
+        - job_name: json
+          metrics_path: /probe
+          params:
+            module: [default]
+          static_configs:
+            - targets:
+              - http://localhost:8000/data.json 
```

Comment:
What is the target the Ops Agent tries to listen on?

`http://localhost:8000/data.json/probe`?

Will it also work if the target is 
`localhost:8000` and `metrics_path` is `/data.json/probe`? 

I ask because this feels like a validation error that is a bug and not a feature lol

Reply:
The final target after the relabeling is: `http://localhost:7979/probe?module=default&target=http://localhost:8000/data.json`. 
Here is what I think how this got built: 
- Initially, `__address__` is the target which is `http://localhost:8000/data.json`;
- The first relabeling move `__address__` to `__param_target`, so that we now have `target=http://localhost:8000/data.json` of the final url;
- The `params: module: [default]` add the `module=default` so we have `module=default&target=http://localhost:8000/data.json`
- The `metric_path: /probe` add that to url and get `probe?module=default&target=http://localhost:8000/data.json`;
- And the last relabeling change the `__address__` to `localhost:7979` and get `http://localhost:7979/probe?module=default&target=http://localhost:8000/data.json`. 

It should also work if target is `localhost:7979` (the JSON exporter's port) and `metric_path` is `/probe`, and params has `module: [default]` and `target: [http://localhost:8000/data.json]`.